cdap-ui/app/directives/metric-picker/metric-picker.js

```diff
@@ -109,45 +145,54 @@ angular.module(PKG.name + '.commons')
         elem.find('button').on('blur', onBlurHandler);
         elem.find('input').on('blur', onBlurHandler);
 
-        scope.$watchCollection('metric', function (newVal, oldVal) {
+        var metricChanged = function (newVal, oldVal) {
           ngModel.$validate();
 
           if(newVal.type !== oldVal.type) {
             scope.metric.context = '';
-            scope.metric.name = '';
+            scope.metric.resetNames();
             fetchAhead();
             return;
           }
 
           if(newVal.context !== oldVal.context) {
-            scope.metric.name = '';
+            scope.metric.resetNames();
             fetchAhead();
+            return;
           }
 
-          if(newVal.context && newVal.name) {
-            if (newVal.name === 'Add All') {
+          if(newVal.context && newVal.names) {
+            var isAddAll = false;
+            for (var i = 0; i < newVal.names.length; i++) {
+              if (newVal.names[i].name === 'Add All') {
+                isAddAll = true;
+              }
+            }
+            if (isAddAll) {
```

Comment:
Probably 'add all' should be a checkbox that removes the metric input boxes. (I agree with the exclusivity of the two).